packages/credential-provider-imds/src/fromInstanceMetadata.e2e.spec.ts

```diff
@@ -0,0 +1,122 @@
+import { CredentialsProviderError } from "@smithy/property-provider";
+import { afterEach, beforeEach, describe, expect, test as it } from "vitest";
+
+import { fromInstanceMetadata, getMetadataToken } from "./fromInstanceMetadata";
+
+describe("fromInstanceMetadata (Live EC2 E2E Tests)", () => {
+  const originalEnv = { ...process.env };
+  let imdsAvailable = false;
+
+  beforeEach(async () => {
+    process.env = { ...originalEnv, AWS_EC2_INSTANCE_PROFILE_NAME: "foo-profile" };
+
+    // Check IMDS availability
+    try {
+      const testProvider = fromInstanceMetadata({ timeout: 1000, maxRetries: 0 });
+      await testProvider();
+      imdsAvailable = true;
+    } catch (err) {
+      imdsAvailable = false;
+    }
+  });
+
+  afterEach(() => {
+    process.env = { ...originalEnv };
+  });
+
+  it("should fetch metadata token successfully", async () => {
+    if (!imdsAvailable) {
+      return;
+    }
+    const options = {
+      path: "/latest/api/token",
+      method: "PUT",
+      timeout: 1000,
+      headers: {
+        "x-aws-ec2-metadata-token-ttl-seconds": "21600",
+      },
+    };
+    const token = await getMetadataToken(options);
+    expect(token).toBeDefined();
+    expect(typeof token).toBe("string");
+    expect(token.length).toBeGreaterThan(0);
+  });
+
+  it("retrieves credentials with account ID on allowlisted instances only)", async () => {
+    if (!imdsAvailable) return;
+
+    const provider = fromInstanceMetadata({ timeout: 1000, maxRetries: 2 });
+    const credentials = await provider();
+
+    expect(credentials).toHaveProperty("accessKeyId");
+    expect(credentials).toHaveProperty("secretAccessKey");
+    expect(typeof credentials.accessKeyId).toBe("string");
+    expect(typeof credentials.secretAccessKey).toBe("string");
+
+    if (!credentials.accountId) {
+      console.log("Skipping account ID test not an allowlisted instance");
+      return;
```

Comment:
make this a separate test and skip the actual test instead of returning from the test